ssoPlugin/control/settings/index.html

```diff
@@ -0,0 +1,194 @@
+		 * property to `true` on all its child form elements, and enable by doing the opposite.
+		 *
+		 * @param {HTMLElement} $form The form element to toggle.
+		 */
+		const toggleForm = ($form) => {
+			const $formElements = $form.querySelectorAll('input, textarea, select, button')
+			if ($form.classList.contains('disabled')) {
+				$form.classList.remove('disabled')
+				$formElements.forEach(($el) => $el.disabled = false)
+			} else {
+				$form.classList.add('disabled')
+				$formElements.forEach(($el) => $el.disabled = true)
+			}
+		}
+
+		// ============================================================================================
+		// APPLICATION METHODS
+		// ============================================================================================
+
+		// Set DOM variables that we'll reference in the following methods thanks to lexical scoping.
+		const $apiSettingsForm = document.getElementById('bif-sso__auth-api-settings')
+		const $apiSettingsFieldDomain = document.getElementById('bif-sso__auth-api-settings__domain')
+		const $apiSettingsFieldClientId = document.getElementById('bif-sso__auth-api-settings__client-id')
+		const $apiSettingsAlertMessage = document.getElementById('bif-sso__auth-api-settings__alert-message')
+
+		/**
+		 * Updates the alert message placeholder with the provided message and type.
+		 *
+		 * Expected types are "error" or "success". Default is "notification" and has no special style.
+		 *
+		 * @param {String}  message      The message that should be displayed.
+		 * @param {String} messageType? The message type, from the list above.
+		 */
+		const alertMessage = (message, messageType = 'notification') => {
+			const filteredClassNames = $apiSettingsAlertMessage.className
+				.split(' ')
+				.filter((className) => !className.startsWith('bif-sso__auth-api-settings__alert-message--'))
+			if (message) filteredClassNames.push(`bif-sso__auth-api-settings__alert-message--${messageType}`)
+			$apiSettingsAlertMessage.className = filteredClassNames.join(' ').trim()
+			$apiSettingsAlertMessage.innerHTML = message
+		}
+
+		/**
+		 * Validates that the required settings exist and aren't falsey.
+		 *
+		 * @param {Object} settings An object containing the settings.
+		 */
+		const areValidSettings = (settings) => {
+			return settings && settings.domain && settings.clientId
+		}
+
+		/**
+		 * Prefills the settings form with the provided values, if they're set.
+		 *
+		 * @param {Object} settings An object containing the settings.
+		 */
+		const prefillSettingsForm = (settings) => {
+			$apiSettingsFieldDomain.value = settings.domain || ''
+			$apiSettingsFieldClientId.value = settings.clientId || ''
+		}
+
+		/**
+		 * Saves the settings to the remote BuildFire datastore.
+		 *
+		 * TODO: Make this a Promise, probably.
+		 *
+		 * @param {Object} settings An object containing the settings.
+		 */
+		const saveSettings = (settings) => {
+			toggleForm($apiSettingsForm)
```

Comment:
Is there any circumstance that we would not want this to disable the form?
I tend to prefer explicit instruction for this kind of thing to prevent weird behaviors from cropping up (e.g. if somehow this got clicked a bunch of times in a row or something)

Reply:
You're totally right! I was being a little too clever by trying to have a single `toggleForm` function handle both enabling and disabling. We would _always_ want this line to be disabling the form (and the next one you flag to be enabling it), so I'll fix. Good catch.